analyzers/src/SonarAnalyzer.CSharp/Rules/SqlKeywordsDelimitedBySpace.cs

```diff
@@ -117,7 +117,9 @@ private sealed class StringConcatenationWalker : SafeCSharpSyntaxWalker
 
             public override void VisitInterpolatedStringExpression(InterpolatedStringExpressionSyntax node)
             {
-                if (TryGetConstantValuesOfInterpolatedStringExpression(node, out var stringParts))
+                if (TryGetConstantValuesOfInterpolatedStringExpression(node, out var stringParts)
+                    && stringParts.Count > 0
+                    && StartsWithSqlKeyword(stringParts[0].Text.Trim()))
```

Comment:
This is the heart of the fix.
The rest are refactorings for precision.

Before, we did not care what the user was interpolated, and we were always raising, e.g. `"{one}{two}"`. Now we are actually checking that the expression contains SQL.